From: Rob Evers <revers@redhat.com>
Date: Thu, 28 Oct 2010 17:53:52 -0400
Subject: [scsi] lpfc: fix mailbox handling for UNREG_RPI_ALL case
Message-id: <1288288434-4069-2-git-send-email-revers@redhat.com>
Patchwork-id: 28968
O-Subject: [RHEL5.6 PATCH 1/3] lpfc: Fix mailbox handling for UNREG_RPI_ALL
	case. (CR: 111249)
Bugzilla: 645881
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

From: Rob Evers on behalf of Emulex <revers@redhat.com>

CR: 111249

https://bugzilla.redhat.com/show_bug.cgi?id=645881

diff --git a/drivers/scsi/lpfc/lpfc_sli.c b/drivers/scsi/lpfc/lpfc_sli.c
index 08faecb..572aa8b 100644
--- a/drivers/scsi/lpfc/lpfc_sli.c
+++ b/drivers/scsi/lpfc/lpfc_sli.c
@@ -1727,7 +1727,8 @@ lpfc_sli_def_mbox_cmpl(struct lpfc_hba *phba, LPFC_MBOXQ_t *pmb)
 	}
 
 	if ((pmb->u.mb.mbxCommand == MBX_UNREG_LOGIN) &&
-	    (phba->sli_rev == LPFC_SLI_REV4))
+	    (phba->sli_rev == LPFC_SLI_REV4) &&
+	    (pmb->u.mb.un.varUnregLogin.rsvd1 == 0x0))
 		lpfc_sli4_free_rpi(phba, pmb->u.mb.un.varUnregLogin.rpi);
 
 	/*
